### PR TITLE
refactor(excel2json): resources.xlsx problem diagnosis

### PR DESCRIPTION
**Pull request by Nora-Olivia-Ammann**

### CODE REVIEW

File: src/dsp_tools/commands/excel2json/models/input_error.py [hunk 1]
```diff
@@ -192,12 +192,52 @@ def execute_error_protocol(self) -> str:
...
+
+        return (
+            f"'classes' is a protected sheet name. Please delete all but one.\n"
+            f"The following sheets contain this in their name:"
+            f"{list_separator}{list_separator.join(self.name_classes)}"
+        )
+
+
+@dataclass
+class ResourceIndividualClasses:
```
Comment (jnussbaum) [inline src/dsp_tools/commands/excel2json/models/input_error.py:217]:
```suggestion
class MissingEntryInClassesSheet:
```

File: src/dsp_tools/commands/excel2json/models/input_error.py [hunk 2]
```diff
@@ -192,12 +192,52 @@ def execute_error_protocol(self) -> str:
         )
 
 
+@dataclass
+class ResourceClassSheet:
+    """This class contains information if the excel contains two sheets with 'classes' in the name."""
```
Comment (jnussbaum) [inline src/dsp_tools/commands/excel2json/models/input_error.py:197]: So, "Englishclasses" or "Frenchclasses" would not be accepted? I think that the only thing which should be forbidden is "classes" itself.

File: src/dsp_tools/commands/excel2json/models/input_error.py [hunk 3]
```diff
@@ -192,12 +192,52 @@ def execute_error_protocol(self) -> str:
         )
 
 
+@dataclass
+class ResourceClassSheet:
```
Comment (jnussbaum) [inline src/dsp_tools/commands/excel2json/models/input_error.py:196]:
```suggestion
class IllegalSheetName:
```

File: src/dsp_tools/commands/excel2json/resources.py [hunk 4]
```diff
@@ -247,24 +252,27 @@ def excel2resources(
     return resources, True
 
 
-def _prepare_classes_df(resource_dfs: dict[str, pd.DataFrame]) -> tuple[pd.DataFrame, dict[str, pd.DataFrame]]:
-    resource_dfs = {k.strip(): v for k, v in resource_dfs.items()}
+def _do_classes_excel_sheet_compliance(resource_dfs: dict[str, pd.DataFrame]) -> list[Problem]:
     sheet_name_list = list(resource_dfs)
+    problems: list[Problem] = []
+    cls_sheet_name = _find_all_classes(sheet_name_list)
```
Comment (jnussbaum) [inline src/dsp_tools/commands/excel2json/resources.py:258]: This is a list of strings, so the variable name cannot be singular

File: src/dsp_tools/commands/excel2json/resources.py [hunk 5]
```diff
@@ -247,24 +252,27 @@ def excel2resources(
...
+    if not cls_sheet_name:
+        problems.append(MissingSheetProblem(missing_sheets=["classes"], existing_sheets=sheet_name_list))
+    elif len(cls_sheet_name) > 1:
+        problems.append(ResourceClassSheet(cls_sheet_name))
+    return problems
+
+
+def _find_all_classes(sheet_name_list: list[str]) -> list[str]:
```
Comment (jnussbaum) [inline src/dsp_tools/commands/excel2json/resources.py:266]: does this find all class sheets? Then it should be renamed to "_find_all_class_sheet_names"

File: src/dsp_tools/commands/excel2json/resources.py [hunk 6]
```diff
@@ -247,24 +252,27 @@ def excel2resources(
...
     cls_sheet_name = [
```
Comment (jnussbaum) [inline src/dsp_tools/commands/excel2json/resources.py:267]: plural

File: src/dsp_tools/commands/excel2json/resources.py [hunk 7]
```diff
@@ -247,24 +252,27 @@ def excel2resources(
...
         ok.group(0) for x in sheet_name_list if (ok := regex.search(r"classes", flags=regex.IGNORECASE, string=x))
```
Comment (jnussbaum) [inline src/dsp_tools/commands/excel2json/resources.py:268]: This is confusing. Why do you search for `r"classes"`? `r"^classes$"` would make more sense to me. I think it should be okay if the substring "classes" appears.

File: src/dsp_tools/commands/excel2json/models/input_error.py (hunk 2 above)
Comment (Nora-Olivia-Ammann) [reply src/dsp_tools/commands/excel2json/models/input_error.py:197]: Yes, I agree this is what i am going to change now. But this was actually already forbidden previously...

File: src/dsp_tools/commands/excel2json/resources.py (hunk 7 above)
Comment (Nora-Olivia-Ammann) [reply src/dsp_tools/commands/excel2json/resources.py:268]: This already existed previously and will be deleted.

**Pull request closed without merging**